src/main/kotlin/app/hashers/CodeLongevity.kt

```diff
@@ -215,15 +218,21 @@ class CodeLongevity(private val serverRepo: Repo,
      * Scans the repo to extract code line ages.
      */
     fun scan() : CodeLineAges? {
+        var storedHead: RevCommit? = null
+        var ageData: CodeLineAges = CodeLineAges()
+
         // Load existing age data if any.
-        val iStream = try { ObjectInputStream(FileInputStream(storagePath)) }
-            catch(e : Exception ) { null }
-        val storedHead = iStream?.readObject() as RevCommit?
-        if (storedHead == head) {
-            return null
+        try {
+            val iStream = ObjectInputStream(FileInputStream(storagePath))
+            val storedHeadId = iStream.readUTF()
```

Comment:
Maybe add comment that this method read first line of file? Really strange naming of method they've made.

Reply:
Not sure whether it read the first line or not, it reads a string whatever format it has. I can add a comment like "Storage format: commit id and CodeLineAges structure following it" if you think it makes sense.

Reply:
Yeah, nice.